rust_snuba/src/processors/mod.rs

```diff
@@ -48,6 +52,45 @@ define_processing_functions! {
     ("GenericGaugesMetricsProcessor", "snuba-generic-metrics", generic_metrics::process_gauge_message),
 }
 
+lazy_static! {
```

Comment:
I think this crate isn't really needed, you could probably just write this as a match statement on the string literals

Reply:
the problem is that hashmap needs to get defined on the heap. so the hashmap needs to get built during runtime. then the elements in the hashmap needs to have the correct lifetimes annotated else rust was complaining about it. 

I looked into how this gets generally solved for rust and there are 2 options. lazy_static or defining a macro. decided to use lazy_static since its already a dependency which is getting pulled in as a transient dependency.

Reply:
you want to use [`OnceCell`](https://doc.rust-lang.org/core/cell/struct.OnceCell.html) here, it is the successor to lazy_static